### PR TITLE
Fix on details about alignment matrix and attention method in MGAN model

### PR DESCRIPTION
**Pull request by GeneZC**
Some interesting observations in my preliminary experiments are that:
1. using dot product to calculate alignment matrix is more efficient.
2. after I modify the details according to the paper, the performance of the model has not been improved as my expectation, instead, it seems that the accuracy and f1-score are dropping. However, this is just in accordance with my knowledge that dot product is better than bi-linear as described in some other papers. 
3. take above just as a story cause' I didn't set seeded initialization properly and the alignment loss depicted in the paper is not implemented.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def _evaluate_acc_f1(self):
                     t_outputs_all = torch.cat((t_outputs_all, t_outputs), dim=0)
 
         test_acc = n_test_correct / n_test_total
-        f1 = metrics.f1_score(t_targets_all, torch.argmax(t_outputs_all, -1), labels=[0, 1, 2], average='macro')
+        f1 = metrics.f1_score(t_targets_all.cpu(), torch.argmax(t_outputs_all, -1).cpu(), labels=[0, 1, 2], average='macro')
```
Comment (songyouwei) [inline train.py]: Do we have to explicitly call .cpu() to migrate to pytorch1.0 ?

Comment (GeneZC) [reply train.py]: Yes, as pytorch1.0 does not allow tensors be operated by numpy methods when the tensor is located on GPU. And sklearn is based on numpy.
